core/domain/classifier_services.py

```diff
@@ -347,6 +349,122 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_training_job_training_data(job_id, training_data):
+    """Checks for the existence of the model and then updates it.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob domain object.
+        training_data: list. The training data of the job.
+
+    Raises:
+        Exception. The ClassifierTrainingJobModel corresponding to the job_id
+            of the ClassifierTrainingJob does not exist.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id, strict=False))
+    if not classifier_training_job_model:
+        raise Exception(
+            'The ClassifierTrainingJobModel corresponding to the job_id of the'
+            'ClassifierTrainingJob does not exist.')
+
+    classifier_training_job = get_classifier_training_job_by_id(job_id)
+    classifier_training_job.update_training_data(training_data)
+    classifier_training_job.validate()
+
+    classifier_training_job_model.training_data = training_data
+    classifier_training_job_model.put()
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
```

Comment:
Docstrings for Args are wrong? It seems to be a list of training jobs whereas you have given exp_id and state_name?